src/main/java/core/basesyntax/Circle.java

```diff
@@ -0,0 +1,37 @@
+package core.basesyntax;
+
+public class Circle extends Figure {
+    private double radius;
+
+    Circle(String color) {
+        super(color);
+    }
+
+    Circle(String color, double radius) {
+        super(color);
+        this.setName(Name.CIRCLE.name());
+        this.radius = radius;
+        this.setArea(calculateArea());
+    }
+
+    @Override
+    public void draw() {
+        String text = "Figure: " + getName() + ", area: " + getArea()
+                + " sq.units, radius: " + getRadius()
+                + " units, color: " + getColor();
+        System.out.println(text);
+    }
+
+    public double getRadius() {
+        return radius;
+    }
+
+    public void setRadius(double radius) {
+        this.radius = radius;
+    }
+
+    @Override
+    public double calculateArea() {
```

Comment:
I can see that part of it is done, but we are still having this `area` field, which strangely enough is not really connected to this method. Overall having 2 sources of the same data, is a bad idea, since you or another programmer can easily calculate the area, set it to the field and then change the radius and call the `getArea()` method, which will produce a problematic situation.
I'd remove that field completely from all of the model classes